pypeit/spectrographs/subaru_focas.py

```diff
@@ -0,0 +1,607 @@
+        """
+        # Definitions for the over scan regions in the DS9 image coordinate.
+        # Format
+        # 1: start of left overscan region
+        # 2: end of left overscan region
+        # 3: start of image region
+        # 4: end of image region
+        # 5: start of right overscan region
+        # 6: end of right overscan region
+        #
+        overscan = {}
+        # binning == 1
+        overscan[1] = np.asarray([
+            # For right (DET_ID == 1) image
+            [2, 8, 9, 520, 521, 536],
+            [537, 552, 553, 1064, 1065, 1071],
+            [1074, 1080, 1081, 1592, 1593, 1608],
+            [1610, 1625, 1626, 2137, 2138, 2143],
+            #
+            # For left (DET_ID == 2) image
+            [2, 8, 9, 520, 521, 536],
+            [537, 552, 553, 1064, 1065, 1071],
+            [1074, 1080, 1081, 1592, 1593, 1608],
+            [1609, 1624, 1625, 2136, 2137, 2142],
+            ])
+
+        # binning == 2
+        overscan[2] = np.asarray([
+            # For right (DET-ID == 1) image
+            [1, 4, 5, 260, 261, 276],
+            [277, 292, 293, 548, 549, 551],
+            [553, 556, 557, 812, 813, 828],
+            [829, 845, 846, 1101, 1102, 1104],
+            #
+            # For left (DET-ID == 2) image
+            [2, 4, 5, 260, 261, 276],
+            [277, 292, 293, 548, 549, 551],
+            [553, 556, 557, 812, 813, 828],
+            [829, 844, 845, 1100, 1101, 1104],
+            ])
+
+        # binning == 4
+        overscan[4] = np.asarray([
+            # For right (DET-ID == 1) image
+            [1, 2, 3, 130, 131, 146],
+            [147, 162, 163, 290, 291, 292],
+            [293, 294, 295, 422, 423, 438],
+            [439, 455, 456, 583, 584, 584],
+            #
+            # For left (DET-ID == 2) image
+            [1, 2, 3, 130, 131, 146],
+            [147, 162, 163, 290, 291, 292],
+            [293, 294, 295, 422, 423, 438],
+            [439, 454, 455, 582, 583, 584],
+            ])
+
+        # Read image
+        msgs.info(f'Attempting to read FOCAS file: {raw_file}, det={det}')
+
+        # NOTE: io.fits_open checks that the file exists
+        hdu_l = io.fits_open(raw_file)
+        head = hdu_l[0].header
+        data = hdu_l[0].data
+
+        # Grab the detector or mosaic parameters
+        # TODO
+        # nimg = 1
+        # mosaic = None
+        detector = self.get_detector_par(det, hdu=hdu_l)
+
+        # Need the exposure time
+        exptime = float(head['EXPTIME'])
+        bin_x = int(head['BIN-FCT1'])
+
+        arr_shape = data.shape
+        # allocate output arrays
+        rawdata = data.astype(float)
+        rawdatasec_img = np.zeros(arr_shape, dtype=int)
+        oscansec_img = np.zeros(arr_shape, dtype=int)
+
+        # collect correct data & overscan for binning in spatial (X) axis
+        oscan_arr = overscan[bin_x][(det - 1)*4:][:4]
+        assert len(oscan_arr) == 4
```

Comment:
This needs an error message.  If a user encounters the AssertionError, I assume that's a coding problem that they need to report.

Reply:
good catch